### ISSUE TITLE
parse commands as commands even if their argument is an unquoted = (or update the doc)

### ISSUE THREAD

**Issue by Camusensei** [koalaman/shellcheck#2022]
#### For bugs
- Rule Id: SC2034
- My shellcheck version: online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2034)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
echo a=b=c | tr = ,

```

#### Here's what shellcheck currently says:
```sh
Line 2:
echo a=b=c | tr = ,
             ^-- SC2034: tr appears unused. Verify use (or export if used externally).
                ^-- SC1068: Don't put spaces around the = in assignments (or quote to make it literal).
```

#### Here's what I wanted or expected to see:
```sh
Line 2:
echo a=b=c | tr = ,
                ^-- SC1068: Don't put spaces around the = in assignments (or quote to make it literal).
```
#### Additional comments
I'm fine with this bug being closed without any action if you think that's the way to go
I'd still like SC2034 not to appear if SC1068 is present though.

**Comment by koalaman** [comment]
ShellCheck now parses assignments according to spec without leniency for spaces around the `=` or for `$` on the left-hand side, and therefore no longer sees this as an assignment (SC1068 is still emitted). Thanks!
